Yoast/wordpress-seo
# Create warning for users who enabled WP core auto updates

⚠️ ⚠️ If this PR gets merged after the first RC for 16.0 has been created, please rebase it on `release/16.0` before merging it. Otherwise there will be a lot of commits to cherry-pick 😅 ⚠️ ⚠️ 

## Context


* If users have auto-updates for WordPress core enabled, this option will cause serious problems when not enabling auto-updates for Yoast SEO. We want to create a warning for these users to act on this.

## Summary


This PR can be summarized in the following changelog entry:

* Adds a notification to prompt users to enable auto-updates for Yoast SEO, in case they also have auto-updates for WordPress Core enabled.

## Relevant technical choices:

* I reused part of the code from `Indexing_Notification_Integration` and `Indexing_Failed_Notification_Presenter`.
* On the Dashboard > Updates page, toggling the WP auto-updates leads to a page refresh. That's why you see the new notification appear immediately. In contrast, on the Plugins page, toggling plugin auto-updates is done by an Ajax request (not a refresh), and therefore you see the new notification only after a page refresh, or when you navigate to another admin page. @JessieHenkes from Product has decided that there is no need to fix this at this point.

## Test instructions

### Test instructions for the acceptance test before the PR gets merged
This PR can be acceptance tested by following these steps:

* Make sure you have major auto-updates for WP Core, and for Yoast SEO, both disabled.
    * For Core, this means you should see "Enable automatic updates for all new versions of WordPress", but not click on it.
    	* You can check if auto-updates for Core are already enabled by going to `wp-admin/update-core.php` of your website. If you see the link 'Switch to automatic updates for maintenance and security releases only.', it means that auto-updates are enabled. Clicking on it should disable auto-updates for Core.
    * For Yoast SEO, this means you should see "Enable auto-updates" but not click on it.
* Refresh the backend, you should NOT see a notification about auto-updating.
* Enable auto-updates for WP Core.
* In the top of the screen, you should see that the number of notifications has been upped with 1.
* It should also say "There is a new notification.", but the implementation of this has been moved to https://yoast.atlassian.net/browse/P2-717. So it's okay if this doesn't work yet in this PR.
* You should see the auto-update notification in the notification centre. It should look like this:

<img width="877" alt="Screenshot 2021-02-15 at 16 41 41" src="https://user-images.githubusercontent.com/20280513/107966840-b106c100-6fac-11eb-948a-b0d4b9e16a06.png">

* The link should bring you to the Plugins page.
* On the Plugins page, enable auto-updates for Yoast SEO.
* After a refresh, you should see the auto-update notification disappear.
* Disable the auto-updates again and see the notification reappear (after a refresh).

_Testing dismissing the notification_
* Dismiss the notification in the notification center.
* Go to the Plugins page, and click on "Enable auto-updates" for Yoast SEO.
* Refresh the page, and see that the notification stays hidden.
* Now click on "Disable auto-updates" and see that the notification is still hidden.
* Go to Dashboard > Updates, and see that the notification is still hidden.
* Turn WP core updates off and on again, and see that the notification reappears.

_Testing on a clean install_
* Auto-updates for WordPress core are enabled by default.
* Verify this on a clean install at Dashboard > Updates.
* See that the notification becomes visible upon activating Yoast SEO.

### Test instructions for QA when the code is in the RC




QA can test this PR by following these steps:

## Impact check

This PR affects the following parts of the plugin, which may require extra testing:

*

## UI changes


## Documentation


## Quality assurance


Fixes https://yoast.atlassian.net/browse/P2-651
